frontend/src/components/smart-contracts/modals/CancelJobModal.tsx

```diff
@@ -0,0 +1,45 @@
+import { useSmartContracts } from 'components/smart-contracts/hooks/useSmartContracts';
+import { TransactionModal } from 'components/smart-contracts/modals/TransactionModal';
+import { ContractReceipt } from 'ethers';
+import { ITransactionModalProps } from 'interfaces/ITransactionModalProps';
+import React from 'react';
+
+interface IProps extends ITransactionModalProps {
+  jobId: string;
+}
+
+export const CancelJobModal = (props: IProps) => {
+  const { jobId, show, onFinish, onConfirmed, onError } = props;
+
+  const { contracts } = useSmartContracts();
+
+  // the logic called to initiate the transaction
+  const callContract = async () => {
+    return contracts.Job.cancelJob(jobId);
+  };
+
+  // what to do when the transaction is confirmed on the blockchain
+  const jobTxConfirmed = (receipt: ContractReceipt) => {
+    const events = receipt.events;
+
+    // extract the JobCanceled event and return the new job id
+    let jobId = '';
+    for (const event of events || []) {
+      if (event.event === 'JobCanceled' && event.args) {
+        jobId = event.args.jobId.toString();
+        break;
+      }
+    }
```

Comment:
No worries here but just want to point out an array method refactor that imho is a little more readable:

```
const jobId = event.filter(event => event.event === 'JobCanceled' && event.args)
  ?.[0]
  ?.args.jobId.toString()
```